app/uk/gov/hmrc/transitmovements/models/values/BytesToHex.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.transitmovements.models.values
+
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import akka.util.ByteString
+
+trait BytesToHex { //{ self: Product with AnyVal =>
+  //def id: ByteString
+
+//  override def toString: String =
+//    s"$productPrefix($hexString)"
+
+  //def hexString = BytesToHex.toHex(id)
+}
+
+object BytesToHex {
+
+  def toHex(bytes: ByteString) = {
```

Comment:
Could make this more functional, e.g. with recursion  
`  def toHexRec(bytes: ByteString, idx: Int = 0, sb: StringBuilder = 
new StringBuilder): String =
 {

 if (idx < bytes.length) 
 toHexRec(bytes, idx + 1, sb.append(f"${bytes(idx)}%02x"))
 else 
 sb.toString
 
 }`

Reply:
Great idea. I have removed the traits altogether.
Did replace this with another algorithm before I noticed your comment